tests/test_super.rs

```diff
@@ -0,0 +1,52 @@
+#![cfg(feature = "macros")]
+
+use pyo3::prelude::*;
+
+#[pyclass(subclass)]
+struct BaseClass {
+    val1: usize,
+}
+
+#[pymethods]
+impl BaseClass {
+    #[new]
+    fn new() -> Self {
+        BaseClass { val1: 10 }
+    }
+
+    pub fn method1(&self) -> usize {
+        self.val1
+    }
+}
+
+#[pyclass(extends=BaseClass)]
+struct SubClass {}
+
+#[pymethods]
+impl SubClass {
+    #[new]
+    fn new() -> (Self, BaseClass) {
+        (SubClass {}, BaseClass::new())
+    }
+
+    fn method2<'a>(self_: PyRef<'_, Self>, py: Python<'a>) -> PyResult<&'a PyAny> {
+        let any: Py<PyAny> = self_.into_py(py);
+        let super_ = any.into_ref(py).py_super()?;
+        super_.call_method("method1", (), None)
```

Comment:
I think it's more interesting if in the test both methods are just called `method`, right? Then it's proving that it's correctly calling the base object's `method` and not infinitely recursing.

Reply:
Done